uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/calender-instance.js

```diff
@@ -0,0 +1,72 @@
+define([
+  'common/moment',
+  'leave-absences/shared/modules/models-instances',
+  'common/models/instances/instance',
+], function (moment, instances) {
+  'use strict';
+
+  instances.factory('CalenderInstance', [
+    'ModelInstance',
+    function (ModelInstance) {
+
+      /**
+       * This method checks whether a date matches the send type.
+       *
+       * @param {string} Date
+       * @param {string} Type of day
+       *
+       * @return {Boolean}
+       * Throws Error if date is not found in calenderData
+       */
+      function checkDate(date, dayType) {
+        var searchedDate = this.calenderData.find(function (data) {
+          return moment(data.date).isSame(moment(date));
+        });
+
+        if (!searchedDate) {
+          throw new Error("Date not found");
+        }
+
+        return searchedDate.type.name === dayType;
+      }
+
+      return ModelInstance.extend({
+
+        /**
+         * This object contains the calendar data.
+         * Default value is empty Array
+         */
+        calenderData: [],
+
+        /**
+         * This method checks whether a date is working day.
+         *
+         * @param {string} Date
+         * @return {Boolean}
+         */
+        isWorkingDay: function (date) {
+          return checkDate.call(this, date, "working_day");
```

Comment:
Is it really necessary to check in `OptionValues` in this case? Because the API already providing us with day type, so it will be extra unnecessary work to check in `OptionValues`. 

Also using `OptionValues` would make the functions `async`.

In `LeaveRequestInstance` it was different, as the API sent only the `id`, so there was no way to know the `status_id` without using the `OptionValues`.